libs/cardano-ledger-core/src/Cardano/Ledger/Core.hs

```diff
@@ -679,3 +680,6 @@ toStrictMaybeL = lens maybeToStrictMaybe (const strictMaybeToMaybe)
 
 fromStrictMaybeL :: Lens' (StrictMaybe a) (Maybe a)
 fromStrictMaybeL = lens strictMaybeToMaybe (const maybeToStrictMaybe)
+
+toProxy :: forall a. a -> Proxy a
+toProxy _ = Proxy
```

Comment:
@lehins `asProxyTypeOf` was not what we needed, so I have defined this `toProxy` here. I'm pretty sure this is not the right place, so let's figure out where it needs to move or if it should be defined in every module where we need it